test-common/ostree-spawn.c

```diff
@@ -587,6 +587,27 @@ ostree_undeploy (GFile *sysroot,
                                         error);
 }
 
+gboolean
+ostree_list_refs_in_repo (GFile      *repo,
+                          CmdResult  *cmd,
+                          GError    **error)
+{
+  g_autoptr(GPtrArray) argv = string_array_new ();
+  CmdArg args[] =
+    {
+      { NULL, OSTREE_BINARY },
+      { NULL, "refs" },
```

Comment:
Do we want to pass the `--collections` argument as well? Would that be useful in the tests?

Reply:
@smspillaz?

Reply:
We could, I'm not sure what purpose that would serve though? As far as I can tell, the only reasons we're using this are:

1) To determine the revision of a flatpak by getting its full ref name then its checksum
2) To figure out which flatpaks are in a repo

I'm not sure where collection refs would come into it?

Reply:
It would give us a bit more assurance that apps are being installed with the correct collection IDs; eventually, I’d want everything to be using collection IDs for uniformity. That said, I left this comment before reviewing the subsequent patch which uses this new function in the ways you describe. I would be happy with pushing it as-is (with the fixup from later in my branch squashed in).

Reply:
Pushed.